shell/browser/electron_browser_client.cc

```diff
@@ -1438,6 +1449,20 @@ void ElectronBrowserClient::CreateWebSocket(
   v8::Isolate* isolate = JavascriptEnvironment::GetIsolate();
   v8::HandleScope scope(isolate);
   auto* browser_context = frame->GetProcess()->GetBrowserContext();
+
+#if BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
+  auto* web_request_api =
+      extensions::BrowserContextKeyedAPIFactory<extensions::WebRequestAPI>::Get(
+          browser_context);
+
+  if (web_request_api && web_request_api->MayHaveProxies()) {
```

Comment:
It looks like `MayHaveProxies` will return true even if there isn't a proxy for this specific URL. So if there's any web request handler registered for any URL, Electron's webRequest API won't get a chance at proxying the websocket.

Can we add a test that does something like:

1. Extension registers a webRequest handler for ws://a.com
2. Electron registers a webRequest handler for ws://b.com
3. Page connects to ws://a.com, extension gets the event
4. Page connects to ws://b.com, Electron gets the event